### PR TITLE
Misc work

### PR DESCRIPTION
**Pull request by est77**
- Fixed OSL bump + pt + nee bug.
- Small metal brdf tweaks,
- Precompute n and k in the metal brdf.

### CODE REVIEW

File: src/appleseed/renderer/kernel/lighting/pt/ptlightingengine.cpp [hunk 1]
```diff
@@ -446,6 +446,22 @@ namespace
                         vertex_aovs);
                 }
 
+                // If we have an OSL shader and this is not the last vertex of the path,
+                // we need to choose one of the closures and set its shading basis into the shading point
+                // for the DirectLightingIntegrator to use it.
+                if (!last_vertex && (m_params.m_enable_dl || m_params.m_enable_ibl))
+                {
+                    const Material::RenderData& material_data =
+                            vertex.m_shading_point->get_material()->get_render_data();
```
Comment (dictoon) [inline src/appleseed/renderer/kernel/lighting/pt/ptlightingengine.cpp]: Fix indentation.

File: src/appleseed/renderer/kernel/shading/shadingcontext.h [hunk 2]
```diff
@@ -138,6 +138,12 @@ class ShadingContext
         const foundation::Color3f&  color,
         const float                 alpha) const;
 
+    // Choose one of the bsdf closures and
```
Comment (dictoon) [inline src/appleseed/renderer/kernel/shading/shadingcontext.h]: Unite both comment lines into a single one.

**Comment by est77** [comment]
Fixed and rebased.

**Comment by dictoon** [comment]
Thanks! Good to merge when Travis is done.

**Pull request merged**